htdocs/libraries/icms/core/Session.php

```diff
@@ -89,144 +87,150 @@ static public function service() {
+	 * 3 - check browser and IP A.B.C, recommended;
+	 * 4 - check browser and IP A.B.C.D;
+	 *
+	 * @var int
+	 * @access public
 	 */
 	public $securityLevel = 3;
 
 	/**
 	 * Security checking level for IPv6 Address types
 	 * Possible value:
-	 *	0 - no check;
-	 *	1 - check browser characteristics (HTTP_USER_AGENT);
-	 *	2 - check browser and IPv6 aaaa:bbbb;
-	 *	3 - check browser and IPv6 aaaa:bbbb:cccc;
-	 *	4 - check browser and IPv6 aaaa:bbbb:cccc:dddd;
-	 *  5 - check browser and IPv6 aaaa:bbbb:cccc:dddd:eeee;
-	 *  6 - check browser and IPv6 aaaa:bbbb:cccc:dddd:eeee:ffff;
-	 *  7 - check browser and IPv6 aaaa:bbbb:cccc:dddd:eeee:ffff:gggg; (recommended)
-	 *  8 - check browser and IPv6 aaaa:bbbb:cccc:dddd:eeee:ffff:gggg:hhhh;
+	 * 0 - no check;
+	 * 1 - check browser characteristics (HTTP_USER_AGENT);
+	 * 2 - check browser and IPv6 aaaa:bbbb;
+	 * 3 - check browser and IPv6 aaaa:bbbb:cccc;
+	 * 4 - check browser and IPv6 aaaa:bbbb:cccc:dddd;
+	 * 5 - check browser and IPv6 aaaa:bbbb:cccc:dddd:eeee;
+	 * 6 - check browser and IPv6 aaaa:bbbb:cccc:dddd:eeee:ffff;
+	 * 7 - check browser and IPv6 aaaa:bbbb:cccc:dddd:eeee:ffff:gggg; (recommended)
+	 * 8 - check browser and IPv6 aaaa:bbbb:cccc:dddd:eeee:ffff:gggg:hhhh;
 	 *
-	 * @var	int
-	 * @access	public
+	 * @var int
+	 * @access public
 	 */
 	public $ipv6securityLevel = 7;
 
 	/**
 	 * Enable regenerate_id
-	 * @var	bool
-	 * @access	public
+	 *
+	 * @var bool
+	 * @access public
 	 */
 	public $enableRegenerateId = false;
 
 	/**
 	 * Constructor
+	 *
 	 * @param object $db reference to the {@link XoopsDatabase} object
-	 * Do we need this $db reference now we're using icms::$xoopsDB?????
+	 *        Do we need this $db reference now we're using icms::$xoopsDB?????
 	 *
 	 */
 	public function __construct(&$db) {
-		$this->db =& $db;
+		$this->db = &$db;
 	}
 
 	/**
 	 * Open a session
-	 * @param	string  $save_path
-	 * @param	string  $session_name
-	 * @return	bool
+	 *
+	 * @param string $save_path
+	 * @param string $session_name
+	 * @return bool
 	 */
 	public function open($save_path, $session_name) {
 		return true;
 	}
 
 	/**
 	 * Close a session
-	 * @return	bool
+	 *
+	 * @return bool
 	 */
-	public function close()	{
+	public function close() {
 		self::gc_force();
 		return true;
 	}
 
 	/**
 	 * Read a session from the database
-	 * @param	string  &sess_id    ID of the session
-	 * @return	array   Session data
+	 *
+	 * @param string &sess_id ID of the session
+	 * @return string Session data
 	 */
 	public function read($sess_id) {
 		return self::readSession($sess_id);
 	}
 
 	/**
 	 * Inserts a session into the database
-	 * @param   string  $sess_id
-	 * @param   string  $sess_data
-	 * @return  bool
-	 **/
+	 *
+	 * @param string $sess_id
+	 * @param string $sess_data
+	 * @return bool
+	 */
 	public function write($sess_id, $sess_data) {
-		return self::writeSession($sess_id, $sess_data);
+		return self::writeSession($sess_id, $sess_data) ? true : false;
```

Comment:
I think `(bool)self::writeSession($sess_id, $sess_data)` here could be a better - inline casting usually is faster than if

Reply:
Applied and committed